File: Max Level Sum in Binary Tree[7-Mar-2023]/Solution.java
class Solution {
    public int maxLevelSum(Node root) {
        
        Queue<Node> q=new LinkedList<>();
        
        q.add(root);
        
        int maxSum=Integer.MIN_VALUE;
        

        while(!q.isEmpty())
        {
            int size=q.size();
            
		//current level sum
            int levelSum=0;
            
            for(int i=0;i<size;i++)
            {
                Node nn=q.poll();
                
                levelSum=levelSum+nn.data;
                
                if(nn.left!=null)
                {
                    q.add(nn.left);
                }
                if(nn.right!=null)
                {
                    q.add(nn.right);
                }
            }
            
		//Maximum Level Sum..
            maxSum=Math.max(maxSum,levelSum);
        }
        
            return maxSum;
    }
}

File: Max Level Sum in Binary Tree[7-Mar-2023]/Solution.cpp
class Solution{
  public:
    int maxLevelSum(Node* root) {

        queue<Node*> q;
            q.push(root);

            int maxSum=INT_MIN;

            while(!q.empty()) {
                int size=q.size();

		    //Current Level Sum..
                int levelSum=0;

                for(int i=0; i<size; i++) {
                    Node* nn=q.front();
                    q.pop();

                    levelSum += nn->data;

                    if(nn->left != NULL) {
                        q.push(nn->left);
                    }
                    if(nn->right != NULL) {
                        q.push(nn->right);
                    }
                }

		   //Maximum Level Sum..
                maxSum=max(maxSum, levelSum);
            }

            return maxSum;
    }
};